### PR TITLE
apm-server 6.0.0 (new formula)

### PR DESCRIPTION
**Pull request by jasontedor**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/apm-server.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
+class ApmServer < Formula
+  include Language::Python::Virtualenv
```
Comment (ilovezfs) [inline Formula/apm-server.rb]: virtualenv isn't needed for installing a build time resource. just use `setup_install_args`

**Review by ilovezfs — CHANGES_REQUESTED** [review]
This should use a resource block and `setup_install_args` instead of the virtualenv DSL.

File: Formula/apm-server.rb (hunk 1 above)
Comment (jasontedor) [reply Formula/apm-server.rb]: `setup_install_args` appears to invoke a `setup.py` and that's not what is going on here?

Comment (ilovezfs) [reply Formula/apm-server.rb]: I meant like this:
```

Comment (ilovezfs) [reply Formula/apm-server.rb]: But actually it looks like you could get away with this:
```
diff --git a/Formula/auditbeat.rb b/Formula/auditbeat.rb
index 6ff7c7e..42a9b29 100644
--- a/Formula/auditbeat.rb
+++ b/Formula/auditbeat.rb
@@ -14,13 +14,11 @@ class Auditbeat < Formula
     ENV["GOPATH"] = buildpath
     (buildpath/"src/github.com/elastic/beats").install buildpath.children
 
-    venv = virtualenv_create(buildpath)
-    venv.pip_install "virtualenv"
-    ENV["PATH"] = ENV["PATH"].split(":").push(buildpath/"bin").join(":")
+    venv = virtualenv_create(buildpath/"venv")
 
     cd "src/github.com/elastic/beats/auditbeat" do
       system "make"
-      system "make", "update"
+      system "make", "update", "PYTHON_ENV=#{buildpath}/venv"
       (libexec/"bin").install "auditbeat"
       libexec.install "_meta/kibana"
 
```

Comment (ilovezfs) [reply Formula/apm-server.rb]: I do prefer the resource block though since it avoids a dependency on the virtualenv DSL.

Comment (jasontedor) [reply Formula/apm-server.rb]: Yet using the virtualenv DSL is clearer and less manual?

Comment (ilovezfs) [reply Formula/apm-server.rb]: Now you know how much I don't like it.

Comment (jasontedor) [reply Formula/apm-server.rb]: Okay, I pushed a new commit.

**Comment by ilovezfs** [comment]
Same `pip install --no-binary :all:` issue here.

**Comment by jasontedor** [comment]
@ilovezfs I have pushed a change.

**Comment by jasontedor** [comment]
Can we pull this in?

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/apm-server.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      exec bin/"apm-server", "-path.config", testpath/"config", "-path.data", testpath/"data"
+    end
+    sleep 1
+
+    begin
+      system "curl", "-H", "Content-Type: application/json", "-XPOST", "localhost:#{port}/v1/transactions", "-d",
+             '{"app":{"name":"app1","agent":{"name":"python","version":"1.0"}},' \
+             '"transactions":[{"id":"945254c5-67a5-417e-8a4e-aa29efcbfb79","name":"GET /api/types","type":"request","duration":32.592981,"timestamp":"2017-05-09T15:04:05.999999Z"}]}'
+      sleep 1
+      s = (testpath/"apm-server/apm-server").read.chomp
```
Comment (ilovezfs) [inline Formula/apm-server.rb]: no need to chomp unless we're using assert_equal unless I've misunderstood something

**Comment by ilovezfs** [comment]
test fails intermittently
```
==> curl -H Content-Type: application/json -XPOST localhost:54824/v1/transactions -d {"app":{"name":"app1","agent":{"name":"python","version":"1.0"}},"transactions":[{"id":"945254c5-67a5-417e-8a4e-aa29efcbfb79","name":"GET /api/types","type":"request","duration":32.592981,"timestamp":"2017-05-09T15:04:05.999999Z"}]}
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   232    0     0  100   232      0  33711 --:--:-- --:--:-- --:--:-- 38666
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/apm-server.rb:99:in `read'
Errno::ENOENT: No such file or directory @ rb_sysopen - /tmp/apm-server-test-20171130-2750-132vl5g/apm-server/apm-server
1. raise
2. backtrace
3. shell
```
maybe sleep is too short?

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
pulling regardless.

**Comment by ilovezfs** [comment]
Thanks for another :new: formula @jasontedor! :rocket:

**Pull request closed without merging**